bundles/org.openhab.binding.mycroft/src/main/java/org/openhab/binding/mycroft/internal/channels/VolumeChannel.java

```diff
@@ -0,0 +1,165 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mycroft.internal.channels;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.mycroft.internal.MycroftBindingConstants;
+import org.openhab.binding.mycroft.internal.MycroftHandler;
+import org.openhab.binding.mycroft.internal.api.MessageType;
+import org.openhab.binding.mycroft.internal.api.dto.BaseMessage;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeDecrease;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeGet;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeGetResponse;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeIncrease;
+import org.openhab.binding.mycroft.internal.api.dto.MessageVolumeSet;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+
+/**
+ * The channel responsible for handling the volume of the Mycroft speaker
+ * NOT FUNCTIONAL
+ * (see https://community.mycroft.ai/t/openhab-plugin-development-audio-volume-message-types-missing/10576)
+ *
+ * @author Gwendal ROULLEAU - Initial contribution
+ */
+@NonNullByDefault
+public class VolumeChannel extends MycroftChannel<State> {
+
+    private PercentType lastVolume = new PercentType(50);
+    private PercentType lastNonZeroVolume = new PercentType(50);
+
+    public VolumeChannel(MycroftHandler handler) {
+        super(handler, MycroftBindingConstants.VOLUME_CHANNEL);
+    }
+
+    @Override
+    public List<MessageType> getMessageToListenTo() {
+        return Arrays.asList(MessageType.mycroft_volume_get_response, MessageType.mycroft_volume_set,
+                MessageType.mycroft_volume_mute, MessageType.mycroft_volume_unmute, MessageType.mycroft_volume_increase,
+                MessageType.mycroft_volume_decrease);
+    }
+
+    @Override
+    public void messageReceived(BaseMessage message) {
+
+        if (message.type == MessageType.mycroft_volume_get_response) {
+            float volumeGet = ((MessageVolumeGetResponse) message).data.percent;
+            updateAndSaveMyState(normalizeVolume(volumeGet));
+        } else if (message.type == MessageType.mycroft_volume_set) {
+            float volumeSet = ((MessageVolumeSet) message).data.percent;
+            updateAndSaveMyState(normalizeVolume(volumeSet));
+        } else if (message.type == MessageType.mycroft_volume_mute) {
+            updateAndSaveMyState(new PercentType(0));
+        } else if (message.type == MessageType.mycroft_volume_unmute) {
+            updateAndSaveMyState(lastNonZeroVolume);
+        } else if (message.type == MessageType.mycroft_volume_increase) {
+            updateAndSaveMyState(normalizeVolume(lastVolume.intValue() + 10));
```

Comment:
Are these numbers fixed in mycroft? I could imagine that one can configure them in mycroft and then openHAB will be out of sync. So maybe you can offer a configuration parameter for the `Thing` where one can set the volume offset for increase and decrease commands.

Reply:
Yes, the increase and decrease step in the volume in Mycroft are fixed.

By the way, I tried to prevent an out of sync volume channel by sending volume.get message periodically (in the periodically called startWebsocket method of the thing handler),
BUT
the volume channel is still broken (as I mention earlier, I didn't have a response to my request on the Mycroft forum asking how it works) and it won't be visible for the openHAB end user.

I let the code here, as you suggest, in case it is clarified some day and then usefull.

Reply:
Fine with me, since you already have a comment with a link to the issue on that class.